doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -926,14 +909,257 @@ the conversion in hypotheses :n:`{+ @ident}`.
       proof term. See `#12200
       <https://github.com/coq/coq/issues/12200>`_ for more details.
 
-Conversion tactics applied to hypotheses
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _controlling-the-reduction-strategies:
+
+Controlling the reduction strategies and the conversion algorithm
+----------------------------------------------------------------------
+
+Coq provides several reduction strategies that tactics can invoke and three
+different algorithms to check the convertibility of types:
+
+- :tacn:`lazy` lazily compares applicative terms.
+- :tacn:`vm_compute` is a brute-force but efficient algorithm that
+  first normalizes the
+  terms before comparing them. It is based on a
+  bytecode representation of terms similar to the bytecode
+  representation used in the ZINC virtual machine :cite:`Leroy90`. It is
+  especially useful for intensive computation of algebraic values, such
+  as numbers, and for reflection-based tactics.
+- :tacn:`native_compute` uses the same algorithm as :tacn:`vm_compute`,
+  but it's based on converting the Coq code to OCaml.
```

Comment:
The tactics do not check convertibility of types.  The tactics fully reduce terms using the same strategy that is used for term comparison